### PR TITLE
Use service instead of controller

### LINKED ISSUE #27: Introduce a Service Layer
The package `de.refactoringbot.controller` and below contains business logic, but the packe and the classes are named ...Controller. In my optinion they should be named `...Service` and use the `@Service` annotation. This change makes it easier to navigate through the code, when the project follows the common layered architecture normally used in spring projects.

Also the RestControllers contain a lot of decission logic. This should also be pushed down into the Service layer to have a better encapsulation of the business logic.

### LINKED ISSUE #28: On first start, let the bot create the DB schema, if it doesn't exist
To ease installation and avoid manual steps, let the bot automatically create the needed DB schema, if it does not exist in the provided MySQL instance. This would of course require the necessary privileges. If this fails, appropriate error handling and log messages should be implemented.

### PR DESCRIPTION
**Pull request by LHommeDeBat**
- Fixes #27, *Introduce a Service Layer*
- Fixes #28, *On first start, let the bot create the DB schema, if it doesn't exist*
- Added env variables with default values inside 'application_example.yml' (#3)

### CODE REVIEW

**Comment by MarvinWyrich** [comment]
Thanks for looking into it.

We should also directly identify possible parts that we should move from the rest controllers to services in order to get the business logic out of the rest controllers.

> Also the RestControllers contain a lot of decission logic. This should also be pushed down into the Service layer to have a better encapsulation of the business logic.

File: src/main/java/de/refactoringbot/rest/ConfigurationController.java [hunk 1]
```diff
@@ -156,43 +117,39 @@
 	@ApiOperation(value = "Delete Git-Configuration with configuration id")
 	public ResponseEntity<?> deleteConfig(@PathVariable("configurationId") Long configurationId) {
 		// Check if configuration exists
-		Optional<GitConfiguration> existsConfig;
+		GitConfiguration config = null;
```
Comment (Simulant87) [inline src/main/java/de/refactoringbot/rest/ConfigurationController.java]: the whole decission logic about how to handle the `GitConfiguration` could also be moved into the `configServie`, so the controller only delegates the call to te `configService.delteConfig(configId)` and handles the result.

File: src/main/java/de/refactoringbot/rest/RefactoringController.java [hunk 2]
```diff
@@ -75,40 +52,26 @@
 	@ApiOperation(value = "Perform refactorings with Pull-Request-Comments.")
 	public ResponseEntity<?> refactorWithComments(@PathVariable Long configID) {
 
-		// Create empty list of refactored Issues
```
Comment (Simulant87) [inline src/main/java/de/refactoringbot/rest/RefactoringController.java]: the whole block of code should be moved into the service layer, to the controller just as to call one entry point and handles the exceptions.

File: src/main/java/de/refactoringbot/rest/RefactoringController.java [hunk 3]
```diff
@@ -251,98 +147,58 @@
 		// Create empty list of refactored Issues
 		Integer amountBotRequests = 0;
 		List<RefactoredIssue> allRefactoredIssues = new ArrayList<>();
-		Optional<GitConfiguration> gitConfig;
```
Comment (Simulant87) [inline src/main/java/de/refactoringbot/rest/RefactoringController.java]: also this whole block of code should be moved into the service layer.

File: src/main/resources/application_example.yml [hunk 4]
```diff
@@ -4,17 +4,17 @@ spring:
   application:
     name: refactoringbot
   datasource:
-    url: jdbc:mysql://localhost:3306/refactoringbot_db?verifyServerCertificate=false&useSSL=true
-    username: root
-    password: root
+    url: jdbc:mysql://${DATABASE_HOST:localhost}:${DATABASE_PORT:3306}/${DATABASE_NAME:refactoringbot_db}?verifyServerCertificate=false&useSSL=true&createDatabaseIfNotExist=true
```
Comment (Simulant87) [inline src/main/resources/application_example.yml]: Auto creation of missing database schema works.

File: src/main/java/de/refactoringbot/rest/ConfigurationController.java (hunk 1 above)
Comment (LHommeDeBat) [reply src/main/java/de/refactoringbot/rest/ConfigurationController.java]: @Simulant87 : I think I addressed everything you requested in your review. Check my latest 2 commits.

**Comment by LHommeDeBat** [comment]
@Simulant87 : I think I addressed everything you requested in your review. Check my latest 2 commits.

**Pull request merged**